### PR TITLE
Fix type check issues

### PR DESCRIPTION
**Pull request by rmitsch**
## Description

Fix type check issues (for LangChain integration).

### Types of change

Bug fix.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran all tests , and all new and existing tests passed. This includes tests marked as `external` and those 
  requiring a GPU. I did this for the following directories:
  - [x] `tests`
  - [x] `usage_example/tests`
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy_llm/backends/integration/langchain.py [hunk 1]
```diff
@@ -21,22 +21,28 @@ def _check_installation() -> None:
 
 
 @registry.llm_queries("spacy.CallLangChain.v1")
-def query_langchain() -> Callable[["BaseLLM", Iterable[Any]], Iterable[Any]]:
+def query_langchain() -> Callable[
+    ["langchain.llms.base.BaseLLM", Iterable[Any]], Iterable[Any]
```
Comment (adrianeboyd) [inline spacy_llm/backends/integration/langchain.py:25]: Don't you need to update the `TYPE_CHECKING` block to correspond to this?

Comment (rmitsch) [reply spacy_llm/backends/integration/langchain.py:21]: I don't think it's necessary at all anymore now?

File: pyproject.toml [hunk 2]
```diff
@@ -15,8 +15,9 @@ testpaths = ["tests"]
 filterwarnings = [
     "error",
     "ignore:^.*pkg_resources.*:DeprecationWarning",
-    "ignore::sqlalchemy.exc.MovedIn20Warning"
+    "ignore::sqlalchemy.exc.MovedIn20Warning",
+    "ignore:^.*You are trying to use a chat model. This way of initializing it is no longer supported. Instead, please use.*:UserWarning",
```
Comment (rmitsch) [inline pyproject.toml:19]: LangChain prefers a different way of initializing OpenAI chat models now. The old one still works however.

Comment (bdura) [reply pyproject.toml:19]: It seems like you did modify the instantiation method though... Should we keep the last exception?

File: spacy_llm/backends/integration/langchain.py [hunk 3]
```diff
@@ -50,14 +52,20 @@ def backend_langchain(
...
+    if "model" not in config:
+        raise ValueError("The LLM model must be specified in the config.")
+
     # langchain.llms.type_to_cls_dict contains all API names in lowercase, so this allows to be more forgiving and make
     # "OpenAI" work as well "openai".
     api = api.lower()
-    type_to_cls_dict: Dict[str, Type[BaseLLM]] = langchain.llms.type_to_cls_dict
+    type_to_cls_dict: Dict[
+        str, Type[langchain.llms.base.BaseLLM]
+    ] = langchain.llms.type_to_cls_dict
```
Comment (bdura) [inline spacy_llm/backends/integration/langchain.py:63]: Why is this necessary?

**Review by bdura — COMMENTED** [review]
Looks good. I think we can remove the error-on-warning exception, and I'm not sure whether re-defining the models dict actually helps - but I trust you.

File: spacy_llm/backends/integration/langchain.py (hunk 3 above)
Comment (rmitsch) [reply spacy_llm/backends/integration/langchain.py:63]: IIRC `mypy` complained otherwise.

File: pyproject.toml (hunk 2 above)
Comment (rmitsch) [reply pyproject.toml:19]: No, I'm just passing `model_name` as separate arg. What LangChain actually wants us to do is to instantiate a differently named class. The current approach does exactly the same behind the LangChain curtains, but with a warning and rerouting to this new class with a warning.

What we should actually do is to figure out a new way how to instantiate the correct `LangChain` classes generically (like it's done currently with `langchain.llms.type_to_cls_dict`. I haven't looked into this and would do so in a separate PR.

File: spacy_llm/backends/integration/langchain.py (hunk 3 above)
Comment (bdura) [reply spacy_llm/backends/integration/langchain.py:63]: Perfect then!

**Review by bdura — APPROVED** [review]
